### ISSUE TITLE
Inconsistent reporting of position number in peptide sequences in pepXML

### ISSUE THREAD

**Issue by mriffle** [smith-chem-wisc/MetaMorpheus#1117]
Greetings, I am running the cross-linking search task. Linked positions in peptides look to be reported such as the first residue is position 1, such as the following looplink at positions 4 and 7--both lysines, which makes sense:

```
      <search_result>
        <search_hit hit_rank="1" peptide="NSGKEQKITIK" peptide_prev_aa="K" peptide_next_aa="A" protein="16128008" num_tot_proteins="1" calc_neutral_pep_mass="1385.79944" massdiff="0.0105552255845112" xlink_type="loop">
          <modification_info />
          <xlink identifier="DSS" mass="138.068085">
            <linked_peptide num_tot_proteins="0" calc_neutral_pep_mass="0" complement_mass="0">
              <xlink_score name="link" value="4" />
              <xlink_score name="link" value="7" />
            </linked_peptide>
          </xlink>
          <search_score name="xlTotalScore" value="6.01665364774797" />
          <search_score name="Qvalue" value="0.032258064516129" />
        </search_hit>
      </search_result>
```

And the following "deadend" is reported correctly for position 3:

```
      <search_result>
        <search_hit hit_rank="1" peptide="LGKHHHHHH" peptide_prev_aa="R" peptide_next_aa="-" protein="H" num_tot_proteins="1" calc_neutral_pep_mass="1297.66577" massdiff="0.00075449932188576" xlink_type="na">
          <modification_info>
            <mod_aminoacid_mass position="3" mass="156.0786" />
          </modification_info>
          <search_score name="xlTotalScore" value="6.09627936326049" />
          <search_score name="Qvalue" value="0.0105386416861827" />
        </search_hit>
      </search_result>
```

However, when reporting any other kind of modification, the numbering seems to be different. For example this fixed/static mod is reported at position 6 when it should be position 5:

```
      <search_result>
        <search_hit hit_rank="1" peptide="MKSECELKEEEIK" peptide_prev_aa="K" peptide_next_aa="A" protein="N_1" num_tot_proteins="1" calc_neutral_pep_mass="1655.81274" massdiff="0.00392896926928188" xlink_type="na">
          <modification_info>
            <mod_aminoacid_mass position="6" mass="57.021463720689994" />
          </modification_info>
          <search_score name="xlTotalScore" value="5.04832156927971" />
          <search_score name="Qvalue" value="0.0280941533788914" />
        </search_hit>
      </search_result>
```

And this variable/dynamic mod is reported at position 2 when it should be position 1:

```
      <search_result>
        <search_hit hit_rank="1" peptide="MTLDDFR" peptide_prev_aa="R" peptide_next_aa="I" protein="P35527" num_tot_proteins="1" calc_neutral_pep_mass="914.4159" massdiff="0.000222794449541652" xlink_type="na">
          <modification_info>
            <mod_aminoacid_mass position="2" mass="15.99491461957" />
          </modification_info>
          <search_score name="xlTotalScore" value="6.04987096923722" />
          <search_score name="Qvalue" value="0.0085580494106249" />
        </search_hit>
      </search_result>
```

Is it that mods are reported such that the n-terminal of the peptide is position 1, so the first position would be 2 and so on? Thanks for any info on this.

**Comment by rmillikin** [comment]
We do refer to the N terminus as the "1" position but this usage is mostly internal. The pepXML output for the XL search is the only place that we output the numbered position; other MetaMorpheus tasks do not output pepXML format yet and our TSV output does not output the location of a mod as a number. Thus the source of confusion.

It's sort of arbitrary whether the N term is referred to as 0 or 1 but of course we should be consistent, thank you for pointing this out. Do you have a preference for this notation? Are you importing the pepXML into a particular program that requires/prefers one over the other?

**Comment by mriffle** [comment]
Ah, that makes sense. I've typically treated the first residue as position 1 (so the n term would be position 0), but honestly either way works for me. I'm writing a converter for MetaMorpheus cross-linking output so it can be used with the proxl viewer and can easily support either choice for reporting the positions.

**Comment by lonelu** [comment]
I will refer position of N terminal mod  as 0 in the crosslink task.

**Comment by rmillikin** [comment]
fixed in #1089 

I'll keep this issue open until we make a new release (probably Monday). Thanks again for the feedback!

**Comment by rmillikin** [comment]
awesome! we're on the lookout for tools to visualize our search results. We're going to revamp our wiki soon-ish; viewing search results will be one of the topics, and we'll aim to include/support proxl.

**Comment by rmillikin** [comment]
fixed in 0.0.278!
